File: 2024-05-25 server/project/src/main/java/com/example/app/model/OrderItem.java
package com.example.app.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "order_items")
public class OrderItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "order_item_id")
    private Long id;

    @ManyToOne
    @JoinColumn(name = "order_id")
    private Order order;

    @ManyToOne
    @JoinColumn(name = "product_id")
    private Product product;

    @Column(name = "quantity")
    private int quantity;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @Column(name = "loaded_quantity")
    private int loadedQuantity;

    // Constructors, getters, setters

    public OrderItem(Order order, Product product, int quantity, User user, int loadedQuantity) {
        this.order = order;
        this.product = product;
        this.quantity = quantity;
        this.user = user;
        this.loadedQuantity = loadedQuantity;
    }
}

File: 2024-05-25/MyActivity2/app/src/main/java/com/example/myactivity/data/model/OrderItem.kt
package com.example.myactivity.data.model

//주문상품 정보
import androidx.constraintlayout.compose.GeneratedValue //import가 안 됨
import androidx.room.Entity


@Entity
@Table(name = "order_items")
data class OrderItem(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "order_item_id")
    var id: Long? = null,

    @ManyToOne
    @JoinColumn(name = "order_id")
    var order: Order? = null,

    @ManyToOne
    @JoinColumn(name = "product_id")
    var product: Product? = null,

    @Column(name = "quantity")
    var quantity: Int = 0,

    @ManyToOne
    @JoinColumn(name = "user_id")
    var user: User? = null,

    @Column(name = "loaded_quantity")
    var loadedQuantity: Int = 0
)
